agent/util-scripts/pbench-tool-meister-start

```diff
@@ -527,150 +616,333 @@ def main(argv):
         else:
-            ret_val = 0
-    else:
-        logger.warning(
-            "unable to successfully start any tool meisters,"
-            " but encountered no failures either: terminating"
-        )
-        ret_val = kill_redis_server(redis_pid)
+            return ReturnCode.TDSWAITFAILURE
+
+    # Setup a Client API object using our existing to_client_chan object to
+    # drive the following client operations ("sysinfo" [optional] and "init"
+    # [required]).
+    with Client(
+        redis_server=redis_server,
+        channel_prefix=cli_tm_channel_prefix,
+        to_client_chan=to_client_chan,
+        logger=logger,
+    ) as client:
+        if sysinfo:
+            sysinfo_path = benchmark_run_dir / "sysinfo" / "beg"
+            try:
+                sysinfo_path.mkdir(parents=True)
+            except Exception:
+                logger.error(
+                    "Unable to create sysinfo-dump directory base path: {}",
+                    sysinfo_path,
+                )
+                ret_val = ReturnCode.EXCSYSINFODIR
+            else:
+                logger.debug("7. Collecting system information")
+                info_log("Collecting system information")
+                ret_val = client.publish(group, sysinfo_path, "sysinfo", sysinfo)
+                ret_val = (
+                    ReturnCode.SUCCESS if ret_val == 0 else ReturnCode.SYSINFOFAILED
+                )
+
+        if ret_val == ReturnCode.SUCCESS:
+            tool_dir = benchmark_run_dir / f"tools-{group}"
+            try:
+                tool_dir.mkdir(exist_ok=True)
+            except Exception as exc:
+                logger.error(
+                    'failed to create tool output directory, "{}": {}', tool_dir, exc
+                )
+                ret_val = ReturnCode.EXCTOOLGROUPDIR
+            else:
+                logger.debug("8. Initialize persistent tools")
+                ret_val = client.publish(group, tool_dir, "init", None)
+                if ret_val != ReturnCode.SUCCESS:
+                    if cli_params.redis_server is None:
+                        # We created the Redis server, so we should clean it up.
+                        ret_val = kill_redis_server(redis_pid, ReturnCode.INITFAILED)
     return ret_val
 
 
+_NAME_ = "pbench-tool-meister-start"
+
 if __name__ == "__main__":
-    status = main(sys.argv)
+    parser = ArgumentParser(
+        f"Usage: {_NAME_} [--sysinfo <list of system information items>]"
+    )
+    parser.add_argument(
+        "--sysinfo",
+        dest="sysinfo",
+        default=None,
+        help="The list of system information items to be collected.",
+    )
+    parser.add_argument(
+        "--redis-server",
+        dest="redis_server",
+        default=os.environ.get("PBENCH_REDIS_SERVER", None),
+        help=(
+            "Use an existing Redis server specified by <hostname>:<port>;"
+            " implies an existing Tool Data Sink and Tool Meisters as well."
```

Comment:
Add help text that specifies that if a `--redis-server` is specified, it is assumed the caller has created and provisioned that Redis server, and the Tool Data Sink and Tool Meisters as well.

Reply:
Add a message about the environment variable as well.

Reply:
Tracked in https://github.com/distributed-system-analysis/pbench/issues/2163.